libraries/stdlib/js/src/js/rtti.js

```diff
@@ -39,22 +39,35 @@ Kotlin.callSetter = function (thisObject, klass, propertyName, value) {
     Kotlin.callSetter(thisObject, Object.getPrototypeOf(klass), propertyName, value);
 };
 
-function isInheritanceFromInterface(ctor, iface) {
-    if (ctor === iface) return true;
-
+function getAllInterfaces(ctor) {
+    if (ctor.$metadata$ == null) ctor.$metadata$ = {};
```

Comment:
would be better to avoid changing `ctor`, and I think we can just return any of `[]`, `null`, `void 0`

Reply:
I don't remember the details. But maybe to not fail on non-kotlin types? But maybe that's handled in the functions calling this one so if that's the case, that line could be removed.

---

To prevent changing the shape of the objects, maybe in the type declaration at the code generation phase, `_allInterfaces` could be set to null so the property at least exists even if computed lazily, or even to an empty array having a boolean elsewhere to determine if it was computed if changing a null still changes the shape.

In a full AOT would be possible to assign numbers to each type beforehand and make a much faster lookup, but having separate modules and being able to depend on super types defined in other module, the lazy option is the best option I thought at that time.